jerry-core/parser/js/js-parser-expr.c

```diff
@@ -353,6 +357,247 @@ parser_append_object_literal_item (parser_context_t *context_p, /**< context */
+      {
+        parser_raise_error (context_p, PARSER_ERR_CLASS_STATIC_PROPERTY_NAME_PROTOTYPE);
+      }
+
+      if (!lexer_check_left_paren (context_p))
+      {
+        lexer_next_token (context_p);
+        parser_raise_error (context_p, PARSER_ERR_LEFT_PAREN_EXPECTED);
+      }
+
+      parser_flush_cbc (context_p);
+
+      uint16_t literal_index = context_p->lit_object.index;
+      uint32_t status_flags = PARSER_IS_FUNCTION | PARSER_IS_FUNC_EXPRESSION | PARSER_IS_CLOSURE;
+      uint16_t function_literal_index = lexer_construct_function_object (context_p, status_flags);
+
+      parser_emit_cbc_literal (context_p,
+                               CBC_PUSH_LITERAL,
+                               function_literal_index);
+
+      JERRY_ASSERT (context_p->last_cbc_opcode == CBC_PUSH_LITERAL);
+
+      context_p->last_cbc.value = literal_index;
+
+      if (is_static)
+      {
+        context_p->last_cbc_opcode = PARSER_TO_EXT_OPCODE (CBC_EXT_SET_STATIC_PROPERTY);
+        is_static = false;
+      }
+      else
+      {
+        context_p->last_cbc_opcode = CBC_SET_LITERAL_PROPERTY;
+      }
+    }
+  }
+
+  if (constructor_literal_p->type == LEXER_UNUSED_LITERAL)
+  {
+    parser_flush_cbc (context_p);
+    constructor_literal_p->u.bytecode_p = parser_create_class_implicit_constructor (context_p);
+    constructor_literal_p->type = LEXER_FUNCTION_LITERAL;
+  }
+
+  JERRY_ASSERT (constructor_literal_p->type == LEXER_FUNCTION_LITERAL);
+} /* parser_parse_class_literal */
+
+/**
+ * Description of "prototype" literal string.
+ */
+static const lexer_lit_location_t lexer_prototype_literal =
+{
+  (const uint8_t *) "prototype", 9, LEXER_STRING_LITERAL, false
+};
+
+/**
+ * Parse class statement or expression.
+ */
+void
+parser_parse_class (parser_context_t *context_p, /**< context */
+                    bool is_statement) /**< true - if class is parsed as a statement
+                                        *   false - otherwise (as an expression) */
+{
+  JERRY_ASSERT (context_p->token.type == LEXER_KEYW_CLASS);
+
+  uint16_t class_ident_index = PARSER_MAXIMUM_NUMBER_OF_LITERALS;
+
+  if (is_statement)
+  {
+    /* Class statement must contain an identifier. */
+    lexer_expect_identifier (context_p, LEXER_IDENT_LITERAL);
+    JERRY_ASSERT (context_p->token.type == LEXER_LITERAL
+                  && context_p->token.lit_location.type == LEXER_IDENT_LITERAL);
+
+    class_ident_index = context_p->lit_object.index;
+    context_p->lit_object.literal_p->status_flags |= LEXER_FLAG_VAR;
+    lexer_next_token (context_p);
+  }
+  else
+  {
+    lexer_next_token (context_p);
+
+    /* Class expression may contain an identifier. */
+    if (context_p->token.type == LEXER_LITERAL && context_p->token.lit_location.type == LEXER_IDENT_LITERAL)
+    {
+      //NOTE: If 'Function.name' will be supported, the current literal object must be set to 'name' property.
```

Comment:
`/* NOTE: If 'Function.name' will be supported, the current literal object must be set to 'name' property. */`

Reply:
Good catch, I haven't noticed it.